src/responder/kcm/kcmsrv_ccache_secdb.c

```diff
@@ -447,20 +455,23 @@ static errno_t secdb_get_cc(TALLOC_CTX *mem_ctx,
         goto done;
     }
 
-    ret = sec_get(tmp_ctx, sreq, &ccbuf);
+    ret = sec_get(tmp_ctx, sreq, &ccbuf, &datatype);
     if (ret != EOK) {
         DEBUG(SSSDBG_OP_FAILURE,
               "Cannot get the secret [%d][%s]\n", ret, sss_strerror(ret));
         goto done;
     }
 
-    ret = sec_kv_to_ccache_json(tmp_ctx, secdb_key,
-                                (const char *)sss_iobuf_get_data(ccbuf),
-                                client, &cc);
+    if (strcmp(datatype, "binary") == 0) {
+        ret = sec_kv_to_ccache_binary(tmp_ctx, secdb_key, ccbuf, client, &cc);
+    } else {
+        ret = sec_kv_to_ccache_json(tmp_ctx, secdb_key,
+                                    (const char *)sss_iobuf_get_data(ccbuf),
+                                    client, &cc);
+    }
     if (ret != EOK) {
-        DEBUG(SSSDBG_OP_FAILURE,
-              "Cannot convert JSON keyval to ccache blob [%d]: %s\n",
-              ret, sss_strerror(ret));
+        DEBUG(SSSDBG_OP_FAILURE, "Cannot convert serialized data to ccache "
```

Comment:
I propose to add `datatype` to this debug message.

Reply:
Done.